packages/property-provider/src/memoize.ts

```diff
@@ -43,16 +43,26 @@ export const memoize: MemoizeOverload = <T>(
   isExpired?: (resolved: T) => boolean,
   requiresRefresh?: (resolved: T) => boolean
 ): Provider<T> => {
+  let result: any;
+  let hasResult: boolean;
   if (isExpired === undefined) {
     // This is a static memoization; no need to incorporate refreshing
-    const result = provider();
-    return () => result;
+    return () => {
+      if (!hasResult) {
+        result = provider();
+        hasResult = true;
+      }
```

Comment:
Remove usage of `hasResult` variable altogether?
```suggestion
      if (!result) {
        result = provider();
      }
```

Ditto on line 62-65

Reply:
This is to catch the cases when result is `null` or `undefined` or `false`